tests/unittests/CoreGraphics.drawing/CGContextDrawing_DrawingOptionsTests.cpp

```diff
@@ -0,0 +1,169 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "DrawingTest.h"
+
+static CGLineCap lineCaps[] = { kCGLineCapButt, kCGLineCapSquare, kCGLineCapRound };
+static CGLineJoin lineJoins[] = { kCGLineJoinBevel, kCGLineJoinRound };
+static CGFloat miterLimits[] = { 1.5, 1.8, 2, 3 };
+struct PhaseDashCount {
+    CGFloat phase;
+    const CGFloat dashPattern[4];
+    size_t count;
+};
+
+struct PhaseDashCount dashParams[] = { { 0, { 5, 2 }, 2 },
+                                       { 5, { 5, 2 }, 2 },
+                                       { 0, { 5, 2, 3, 1 }, 4 },
+                                       { 0, { 1, 1 }, 2 },
+                                       { 1, { 1, 1 }, 2 } };
+
+class CGDrawLineCaps : public WhiteBackgroundTest<>, public ::testing::WithParamInterface<CGLineCap> {
+    CFStringRef CreateOutputFilename() {
+        CGLineCap lineCap = GetParam();
+
+        const char* lineCapName;
+        switch (lineCap) {
+            case kCGLineCapButt:
+                lineCapName = "LineCapButt";
+                break;
+            case kCGLineCapSquare:
+                lineCapName = "LineCapSquare";
+                break;
+            case kCGLineCapRound:
+                lineCapName = "LineCapRound";
+                break;
+            default:
+                break;
+        }
+
+        return CFStringCreateWithFormat(nullptr, nullptr, CFSTR("TestImage.LineWithCap.%s.png"), lineCapName);
+    }
+};
+
+class CGDrawLineJoins : public WhiteBackgroundTest<>, public ::testing::WithParamInterface<CGLineJoin> {
+    CFStringRef CreateOutputFilename() {
+        CGLineJoin lineJoin = GetParam();
+
+        const char* lineJoinName;
+        switch (lineJoin) {
+            case kCGLineJoinMiter:
+                lineJoinName = "LineJoinMiter";
+                break;
+            case kCGLineJoinBevel:
+                lineJoinName = "LineJoinBevel";
+                break;
+            case kCGLineJoinRound:
+                lineJoinName = "LineJoinRound";
+                break;
+            default:
+                break;
+        }
+
+        return CFStringCreateWithFormat(nullptr, nullptr, CFSTR("TestImage.LineWithJoin.%s.png"), lineJoinName);
+    }
+};
+
+class CGDrawLineMiter : public WhiteBackgroundTest<>, public ::testing::WithParamInterface<CGFloat> {
+    CFStringRef CreateOutputFilename() {
+        CGFloat miterLimit = GetParam();
+
+        return CFStringCreateWithFormat(nullptr, nullptr, CFSTR("TestImage.LineWithMiter.%f.png"), miterLimit);
```

Comment:
For all %f's, truncate them to 1 decimal place. No need for obscene file name lengths.